### PR TITLE
Adding information about updating developer environment to the developer docs

### LINKED ISSUE #1181: add information about updating dev environment to dev docs
Under "If you’re having trouble installing or using the development environment" in the dev docs, it would be nice to have the right commands for updating.

For rust, this seems to be:

```
rustup update
```

for python, it varies, I think? you would do
```
pip install -e .
```
and/or
```
conda update --all
```
I think?

### PR DESCRIPTION
**Pull request by keyabarve**
* This involves modifications to `doc/developer.md`. 

* A new heading called "Updating your developer environment" has been added just above "Running tests and checks" in `doc/developer.md`. The commands for updating Rust and Python have been added under this heading.

* There was a typo in the current dev docs on the line:

> Finally, ou can also explicitly install all the Python dependencies for sourmash by running

'ou' has been modified to 'you' here.

* Fixes #1181

### CODE REVIEW

**Comment by keyabarve** [comment]
Please review @ctb @luizirber

File: doc/developer.md [hunk 1]
```diff
@@ -64,12 +64,18 @@ tox -e dev
...
-Finally, ou can also explicitly install all the Python dependencies for sourmash by running
+Finally, you can also explicitly install all the Python dependencies for sourmash by running
 ```
 pip install -r requirements.txt
 ```
 (but they are already installed in the virtualenv created with `tox -e dev`).
 
+## Updating your developer environment
+
+To update rust to the latest version, use [`rustup update`].
```
Comment (ctb) [inline doc/developer.md]: curious about the use of square brackets here - is there a reason to go with this convention? I don't think we use it anywhere else in the code base.

File: doc/developer.md [hunk 2]
```diff
@@ -64,12 +64,18 @@ tox -e dev
...
+
+To update your Python dependencies to the latest required for sourmash, you can run [`pip install -r requirements.txt`].
```
Comment (ctb) [inline doc/developer.md]: same here.

**Review by ctb — CHANGES_REQUESTED** [review]

File: doc/developer.md (hunk 1 above)
Comment (keyabarve) [reply doc/developer.md]: I think I must have put it there by mistake. I'll remove it right away.

**Comment by keyabarve** [comment]
Please review @ctb @luizirber

**Comment by ctb** [comment]
Thank you!

**Review by ctb — APPROVED** [review]

**Comment by ctb** [comment]
@keyabarve can you merge? does the 'squash and merge' button appear for you? I'd be happy to have you do the honors :)

**Comment by keyabarve** [comment]
@ctb I can see the 'squash and merge' button right below. Should I click on it to merge? Please let me know. Thanks!

**Pull request merged**